### PR TITLE
chore: scope third-party warning suppressions to their own sources

### PR DESCRIPTION
**Pull request by ibmibmibm**
## Description

> **Ready for review.** CI is green (`codecov/patch` is the only red — a coverage-delta on two plugin lines that execute only under a loaded model, which the coverage suite does not run; `codecov/project` passes). The wasi_nn surface here is worth a careful look — see **What CI still has to decide** and the two flagged items at the end.

Follow-up to #5131, which upgraded simdjson to v4.6.4 and exposed simdjson's interface include directories as **system includes** (`-isystem`, or `-external:I` on MSVC).

The two per-compiler suppression lists on the `simdjson` target are attached as `PUBLIC`. CMake therefore appends them to the command line of **every source of every target that links simdjson** — not merely to files that `#include` it. That silently disabled 17 warnings (`-Wno-sign-conversion`, `-Wno-old-style-cast`, `-Wno-implicit-int-conversion`, `/wd4100`, …) across `test/spec/spectest.cpp` and the whole `plugins/wasi_nn` target, for those targets' **own code**. That was never the intent.

This PR narrows both lists to `PRIVATE`. Now that consumers reach simdjson's headers through `-isystem`, the lists only have to cover simdjson's own sources, which they still do.

### What the leak was hiding

Removing it surfaced three real problems. All three are invisible on Linux, because the warnings involved are not in `-Wall -Wextra`. They are visible on the Windows job because **clang-cl treats `-Wall` as MSVC's `/Wall`, which expands to `-Weverything`**. (That is also why `WASMEDGE_CFLAGS` carries `-Weverything`-only suppressions such as `-Wno-c++98-compat` and `-Wno-padded`, even though `-Weverything` appears nowhere in the CMake.)

**1. `spectest.cpp` bound loop variables to temporaries** — six sites:

```cpp
for (const simdjson::dom::object &Cmd : CmdArray) {   // binds to a temporary
```

`simdjson::dom::array` iteration yields elements **by value**, so the `const&` binds to a temporary (`-Wrange-loop-bind-reference`). `dom::object` is a lightweight handle (a tape pointer plus an index), so the reference bought nothing and misstated the lifetime. Fixed by iterating by value.

**2. Two unchecked narrowings in the wasi_nn plugin** (`-Wsign-conversion`):

```
plugins\wasi_nn\wasinnenv.cpp(58,15):            'streamoff' -> 'size_type'
plugins\wasi_nn\GGML\core\ggml_core.cpp(130,46): 'size_type' -> 'std::streamsize'
```

Neither file includes simdjson. They were covered only because their *target* links it. Fixed with explicit `static_cast`s, matching the surrounding style.

**3. llama.cpp's headers were compiled under our warning flags.** `common/base64.hpp` reaches the plugin's own translation units through plain `-I`, so it was compiled with `WASMEDGE_CFLAGS` and survived `-Werror` only because simdjson leaked `-Wno-sign-conversion` and `-Wno-implicit-int-conversion` onto the plugin target. Eight errors, in a vendored third-party header we do not control.

The fix is the same one #5131 applied to simdjson: mark llama.cpp's interface include directories as system includes, so consumers get `-isystem` / `-external:I`. Since two dependencies now need this, it is factored into `wasmedge_mark_system_includes()`. (`INTERFACE_SYSTEM_INCLUDE_DIRECTORIES` rather than CMake 3.25's `SYSTEM` property, because the floor here is 3.18.)

The existing `-Wno-error=` block for llama.cpp is left untouched — this PR does not try to prove it dead.

### Verified

Locally on GCC 15.2.0 and Clang 21.1.8: `simdjson` and both consuming test suites build with zero warnings under `-Wall -Wextra -Werror`, and both suites pass.

Scoping was checked against `compile_commands.json` rather than assumed:

| TU | simdjson suppression flags | `-isystem` simdjson |
|---|---|---|
| `simdjson.cpp` (own source) | all retained | no |
| `spectest.cpp` (consumer) | none | yes |

The `-Wno-` flags still on `spectest.cpp` come from `test/CMakeLists.txt`, not from simdjson.

Because Linux cannot see these warnings, each consumer was recompiled with all 17 formerly-leaked warnings re-enabled:

- `spectest.cpp`: 6 × `-Wrange-loop-bind-reference` before the fix, **0** after.
- `wasinnenv.cpp`, `wasinnfunc.cpp`, `wasinnmodule.cpp`, `ggml_core.cpp`, `GGML/utils.cpp`, `GGML/metadata/metadata_parser.cpp`: **0 errors, 0 warnings** after the fixes.
- `base64.hpp` reached from `ggml_core.cpp`: **12** warnings with llama on `-I`, **0** with llama on `-isystem`. Verified both directions, so the suppression is attributable to the change rather than to the probe missing the header.

No warning is attributed to any simdjson or llama.cpp header, confirming the system includes do their job.

### CI history on this branch

| Run | Change | Result |
|---|---|---|
| 1 | delete both suppression lists | `Windows` + `Windows-MSVC` red — the lists are load-bearing for simdjson's own sources |
| 2 | `PRIVATE` + `spectest.cpp` loop fix | `Windows`, `Windows-MSVC`, all `macOS` **green**; `WASI-NN (Windows)` red on `wasinnenv.cpp` |
| 3 | + `wasinnenv.cpp` cast | 92 checks green; `WASI-NN (Windows)` red on `base64.hpp` + `ggml_core.cpp` |
| 4 | + llama system includes, `ggml_core.cpp` cast | 107 green; whole plugin backend matrix passed; `WASI-NN (Windows)` red on 51 masked `-Wsign-conversion` sites in GGML's own sources |
| 5 | + `-Wno-error=sign-conversion` / `extra-semi-stmt` on the plugin target | `WASI-NN (Windows)` **green**; whole matrix green |
| 6 | + negative-`tellg()` guard in `load()` (Copilot review) | green (one OpenVINO clang job flaked, passed on rerun) |

### Reviewer notes — two things worth a close look

1. **`-Wno-error=sign-conversion` / `-Wno-error=extra-semi-stmt` on the wasi_nn target.** These demote (do not silence) 51 real sites in GGML's own sources, 40 of them `int`-indexed `std::vector` access in `GGML/tts/tts_core.cpp`'s FFT routines. They keep the build green without reworking signed index arithmetic in code that cannot be runtime-tested here. A follow-up should burn these down and remove the two flags.
2. **llama.cpp `-isystem` treatment.** `wasmedge_mark_system_includes()` re-exports llama's interface include dirs as system includes so its headers stop compiling under `-Werror`. This is the same fix #5131 applied to simdjson; the existing `-Wno-error=` block for llama is left untouched rather than assumed dead.

### The GGML sign-conversion sites (run 4 → 5)

Run 4 turned the entire plugin backend matrix green — whisper, piper, mlx, chattts, openvino, pytorch, bitnet, tensorflow, across manylinux / ubuntu / darwin — and confirmed no warning comes from any llama.cpp or simdjson header. The only remaining failure was `WASI-NN (Windows)`, and every error was now in **WasmEdge's own GGML sources**: 51 unique `-Wsign-conversion` / `-Wextra-semi-stmt` sites, 40 of them in `GGML/tts/tts_core.cpp`, which index `std::vector` with `int` loop counters in FFT routines.

These are the last thing the leak was hiding. Fixing all 51 means reworking signed index arithmetic in DSP code, which does not belong in a build-scoping change. Instead they are added to the plugin target's **existing** `-Wno-error=` block — the one already carrying `-Wno-error=unused-function`, `-Wno-error=documentation`, `-Wno-error=unused-template` for llama.cpp. `-Wno-error=` demotes without silencing: the diagnostics still print, the build no longer fails. Verified on the configured `tts_core.cpp` command line — 50 warnings / 0 errors with the flags, 50 errors / exit 1 without — and a full re-sweep of all 23 plugin TUs now passes with `-Werror` still on. A follow-up should burn the sites down and remove the two flags.

Still unverified locally: AppleClang and CI's clang 15 / 16 / 18, though all `macOS` jobs passed from run 2 onward.

> This contribution was developed with assistance from Claude (Anthropic). The commit carries the `Assisted-by: Claude (Anthropic)` trailer, and a human verified the change and test results.

## Checklist

Before submitting this PR, please ensure the following:

- [x] **DCO Signed-off**: All commits are signed-off (`git commit -s`). CI workflows will only be approved if DCO check is passed.
- [x] **Commit Messages**: Run `commitlint` on your commit messages to ensure they meet the [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standards.
- [x] **Coding Style**: Run `clang-format` on your code to ensure it meets the coding style guidelines.
- [x] **Local Tests Passed**: Provide a screenshot or logs below to prove that you have passed the test suites locally.

If you are using AI-assisted tools, please ensure the following:

- [x] **AI Disclosure**: Disclose it using the `Assisted-by:` trailer in your commits and note it in the PR description. Please refer to the CONTRIBUTING.md/AGENTS.md files for the detailed format.
- [x] **Human-in-the-loop**: Submitting commits, issues, and PRs without human verification is forbidden.

## Test Evidence

```
$ g++ --version | head -1
g++ (Ubuntu 15.2.0-16ubuntu1) 15.2.0
$ cmake --build build --target simdjson wasmedgeExecutorCoreTests wasmedgeComponentRegressionTests
gcc: warnings = 0, errors = 0
$ ctest -R 'wasmedgeExecutorCoreTests|wasmedgeComponentRegressionTests'
100% tests passed, 0 tests failed out of 2

$ clang++ --version | head -1
Ubuntu clang version 21.1.8 (6ubuntu1)
$ cmake --build build-clang --target simdjson wasmedgeExecutorCoreTests wasmedgeComponentRegressionTests
clang: warnings = 0, errors = 0
$ ctest -R 'wasmedgeExecutorCoreTests|wasmedgeComponentRegressionTests'
100% tests passed, 0 tests failed out of 2

$ clang-format --dry-run -Werror test/spec/spectest.cpp plugins/wasi_nn/wasinnenv.cpp \
    plugins/wasi_nn/GGML/core/ggml_core.cpp    # exit 0
$ lineguard --config .lineguardrc <all 5 changed files>
✓ All files passed lint checks!
```

wasi_nn plugin TUs recompiled with the 17 formerly-leaked warnings re-enabled (GGML backend configured, llama.cpp fetched):

```
  utils.cpp             errors=0 warnings=0
  wasinnenv.cpp         errors=0 warnings=0
  wasinnfunc.cpp        errors=0 warnings=0
  wasinnmodule.cpp      errors=0 warnings=0
  ggml_core.cpp         errors=0 warnings=0
  metadata_parser.cpp   errors=0 warnings=0

base64.hpp warnings with llama on -I:       12
base64.hpp warnings with llama on -isystem:  0
```

(The two `-Wimplicit-int-conversion` hits in `include/common/int128.h` seen during probing are a Linux-only artifact of the `unsigned __int128` path; core lib TUs include that header on Windows without simdjson's flags and are green, so it cannot regress there.)

### CODE REVIEW

File: plugins/wasi_nn/wasinnenv.cpp [hunk 1]
```diff
@@ -55,7 +55,7 @@ bool load(const std::filesystem::path &Path, std::vector<uint8_t> &Data) {
   File.seekg(0, std::ios::end);
   std::streampos FileSize = File.tellg();
   File.seekg(0, std::ios::beg);
-  Data.resize(FileSize);
+  Data.resize(static_cast<size_t>(FileSize));
   File.read(reinterpret_cast<char *>(Data.data()), FileSize);
```
[the remark this answers is not in the document]
Comment (ibmibmibm) [reply plugins/wasi_nn/wasinnenv.cpp]: Good catch — fixed in 332752629. Added a `FileSize < 0` guard that returns early on a failed `tellg()` (previously `Data.resize()` would have received `SIZE_MAX` and thrown), and made the `File.read()` size an explicit `static_cast<std::streamsize>`. Note this was pre-existing rather than introduced here: the prior code was `Data.resize(FileSize)`, which already narrowed a possible -1 implicitly; this PR only made that cast explicit, and now guards it.

**Review by hydai — APPROVED** [review]
